src/videojs-hls.js

```diff
@@ -321,12 +321,21 @@ videojs.Hls.prototype.play = function() {
     this.mediaIndex = 0;
   }
 
-  // seek to the latest safe point in the media timeline when first
-  // playing live streams
-  if (this.duration() === Infinity &&
-      this.playlists.media() &&
-      !this.player().hasClass('vjs-has-started')) {
-    this.setCurrentTime(this.seekable().end(0));
+  // we may need to seek to begin playing safely for live playlists
+  if (this.duration() === Infinity) {
+
+    // if this is the first time we're playing the stream or we're
+    // ahead of the latest safe playback position, seek to the live
+    // point
+    if (!this.player().hasClass('vjs-has-started') ||
+        this.currentTime() > this.seekable().end(0)) {
```

Comment:
I don't believe calculating seekable end is expensive enough to worry about caching.